test/bats/tests/gcp/gcp_setup.bash

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+gcp_project=${GCP_PROJECT_NAME:-"gcp-e2e-test2"}
+sa_account_path="test/bats/tests/gcp/gcpsajson.json"
+secret_name="test-secret-a"
+service_account=${SERVICE_ACCOUNT:-"gcp-test"}
+
+gcloud config set project $gcp_project
+
+
+#Check if the key already exsits by retrieving it 
+
+if [ ! -f "$sa_account_path" ]; then
+  
+    gcloud iam service-accounts keys create --iam-account $service_account@$gcp_project.iam.gserviceaccount.com $sa_account_path
+
+    if [ $? -ne 0 ]; then
+        echo "Error: Cannot export GCP Service Account (GCP_SA_JSON)"
+        return 1
+    fi
+
+    gcloud services enable secretmanager.googleapis.com
+
+    printf "hunter2" | gcloud secrets create $secret_name --replication-policy="automatic" --data-file="-"
```

Comment:
If this script creates the secret then it should probably also export RESOURCE_NAME at the end too in the format:

```
projects/$gcp_project/secrets/$secret_name/versions/latest
```